apps/web/ee/pages/api/cron/workflows/scheduleEmailReminders.ts

```diff
@@ -0,0 +1,134 @@
+/* Schedule any workflow reminder that falls within 72 hours for email */
+import { WorkflowActions, WorkflowTemplates, WorkflowMethods } from "@prisma/client";
+import client from "@sendgrid/client";
+import sgMail from "@sendgrid/mail";
+import dayjs from "dayjs";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import emailReminderTemplate from "@ee/lib/workflows/reminders/templates/emailReminderTemplate";
+
+import prisma from "@lib/prisma";
+
+const sendgridAPIKey = process.env.SENDGRID_API_KEY as string;
+const senderEmail = process.env.SENDGRID_EMAIL as string;
+
+sgMail.setApiKey(sendgridAPIKey);
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const apiKey = req.headers.authorization || req.query.apiKey;
+  if (process.env.CRON_API_KEY !== apiKey) {
+    res.status(401).json({ message: "Not authenticated" });
+    return;
+  }
+
+  if (req.method !== "POST") {
+    res.status(405).json({ message: "Invalid method" });
+    return;
+  }
```

Comment:
Prefer splitting each method logic in their own files and use defaultHandler to auto-handle missing methods. And keep legibility when checking for individual method logic.

📄 [From Cal.com's GitBook](https://app.gitbook.com/o/6snd8PyPYMhg0wUw6CeQ/s/VXRprBTuMlihk37NQgUU/engineering/best-practices/prefer-defaulthandler-for-simple-api-handlers)

__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix